sql/schema_changer.go

```diff
@@ -54,3 +59,77 @@ func (p *planner) applyMutations(tableDesc *TableDescriptor) error {
 	p.notifyCompletedSchemaChange(newTableDesc.ID)
 	return nil
 }
+
+// TableLeaderLeaser is used to acquire a table leader lease on a table.
+type TableLeaderLeaser struct {
+	p *planner
+}
+
+// NewTableLeaderLeaserForTesting for tests.
+func NewTableLeaderLeaserForTesting(txn *client.Txn) TableLeaderLeaser {
+	return TableLeaderLeaser{p: &planner{txn: txn}}
+}
+
+var errExistingTableLeader = errors.New("an outstanding lease exists")
+
+// Acquire acquires a leader lease on the table if
+// an unexpired lease doesn't exist. It returns the lease.
+func (t TableLeaderLeaser) Acquire(tableID ID) (TableDescriptor_LeaderLease, error) {
+	t.p.txn.SetSystemDBTrigger()
+	tableDesc, err := t.p.getTableDescFromID(tableID)
+	if err != nil {
+		return TableDescriptor_LeaderLease{}, err
+	}
+
+	now := time.Now()
+	// Add 5 seconds to the lease expiration to deal with time uncertainty across nodes.
```

Comment:
The 5 seconds should perhaps be a multiple of the configured MaxOffset. Is there a way to use database time (via a `Deadline` on all `EndTransactionRequests` issued under the authority of this lease) to avoid any question of clock offsets?